client/js/utils/utils.js

```diff
@@ -0,0 +1,32 @@
+var _ = {};
+
+_.each = function(collection, callback){
+  if(collection instanceof Array){
+    for(var i = 0; i < collection.length; i++){
+      callback(collection[i], i, collection);
+    }
+  }else{
+    for(var key in collection){
+      callback(collection[key], key, collection);
```

Comment:
add a check - collection.hasOwnProperty(key) to make sure we don't iterate over any enumerable properties added up the prototype chain